solidity/contracts/utility/MocBTCToUSDOracle.sol

```diff
@@ -0,0 +1,55 @@
+pragma solidity 0.4.26;
+
+import "./interfaces/IConsumerPriceOracle.sol";
+import "./Owned.sol";
+
+interface Medianizer {
+    function peek() external view returns (bytes32, bool);
+}
+
+contract MocBTCToUSDOracle is IConsumerPriceOracle, Owned {
+
+    address public mocOracleAddress;
+
+    event SetMoCOracleAddress(address indexed mocOracleAddress, address changerAddress);
+
+    /**
+      * @dev initializes a ne MoC oracle
+      *
+      * @param _mocOracleAddress MoC oracle address
+    */
+    constructor(address _mocOracleAddress) public {
+        setMoCOracleAddress(_mocOracleAddress);
+    }
+
+    /**
+      * @dev returns the USD/BTC rate.
+      *
+      * @return always returns the rate of 10000
+    */
+    function latestAnswer() external view returns (int256) {
+        (bytes32 value, bool hasValue) = Medianizer(mocOracleAddress).peek();
+        require(hasValue, "Doesn't has value");
+        return int256(value);
+    }
+
+    /**
+      * @dev returns the USD/BTC update time.
+      *
+      * @return always returns current block's timestamp
+    */
+    function latestTimestamp() external view returns (uint256) {
+        return now; // MoC oracle doesn't return update timestamp
```

Comment:
@ororopickpocket what do you think about this? Could this be a problem in the future?

Reply:
We don't have much of a choice here if MoC isn't returning the timestamp...
Where and for what is latestTimestamp() used by the protocol?

Reply:
It is used in the PriceOracle to get the last updated price timestamp

Reply:
update: the AMM is designed to use the internal rate if the price hasn't been updated more recently. With the changes introduced in this PR, we're always using the rate returned by the oracle for weight balancing. According to MoC the price is updated every 10 minutes or every time the price moves 0.1%, so it should be ok.